src/core/Track.cpp

```diff
@@ -735,8 +735,24 @@ void TrackContentObjectView::paintTextLabel(QString const & text, QPainter & pai
  */
 void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 {
-	// List of selected TCOs
-	QVector<selectableObject *> so = gui->songEditor()->m_editor->selectedObjects();
+	// Get a list of selected selectableObjects
+	QVector<selectableObject *> sos = gui->songEditor()->m_editor->selectedObjects();
+
+	// Convert to a list of selected TCOVs
+	QVector<TrackContentObjectView *> selection;
+	selection.reserve( sos.size() );
+	for( auto so: sos )
+	{
+		TrackContentObjectView *tcov = dynamic_cast<TrackContentObjectView *> ( so );
+		if( tcov != nullptr )
+		{
+			selection.append( tcov );
+		}
+	}
+
+	// If we clicked part of the selection, affect all selected clips. Otherwise affect the clip we clicked
+	QVector<TrackContentObjectView *> active = selection.contains( this ) ? selection : QVector<TrackContentObjectView *>( 1, this );
```

Comment:
Let's extract this into a new method (`getClickedTCOs`?) since you reuse the same code later.

Reply:
Good idea! I updated the code to use this new method, which I'm calling `getClickedTCOs` as suggested for now